packages/gatsby/src/commands/build-html.ts

```diff
@@ -314,12 +315,22 @@ export const doBuildPages = async (
   try {
     await renderHTMLQueue(workerPool, activity, rendererPath, pagePaths, stage)
   } catch (error) {
+    const pageData = await getPageData(error.context.path)
+
+    // These objs can be quite large, so we want to limit the character count
+    const pageDataMessage = `Data for failed page "${
+      error.context.path
+    }": ${JSON.stringify(pageData, null, 2).slice(0, 50000)}`
```

Comment:
an additional aid for this would be to iterate over every key on the object and [truncate](https://lodash.com/docs/4.17.15#truncate) the string to e.g. 300 characters. That'd keep almost any data result to less than 50k characters then. Which is nice as the main goal here is that people can see which fields are either missing or of the wrong type.

Reply:
Good idea, I truncated the values in `pageData` in the latest commit.